frontend/src/common/redux/userLogin.ts

```diff
@@ -208,40 +209,57 @@ export const registerViaEmailAndPassword = createAsyncThunk<
   {
     email: string;
     password: string;
-    displayName: string;
+    passwordConfirmation: string;
+    firstName: string;
+    lastName: string;
     recaptcha: string | null;
   },
   ThunkApiType
 >(
   "currentUser/registerViaEmailAndPassword",
-  async ({ email, password, displayName, recaptcha }, thunkAPI) => {
+  async ({ email, password, passwordConfirmation, firstName, lastName, recaptcha }, thunkAPI) => {
     if (!recaptcha) {
       return thunkAPI.rejectWithValue({
-        message: "Please complete the recaptcha",
+        message: "Please complete the recaptcha"
       });
     }
-    if (email === "") {
+    if (!email || email === "") {
       return thunkAPI.rejectWithValue({
-        message: "Please enter your email",
+        message: "Please enter your email"
       });
     }
-    if (password === "") {
+    if (!password || password === "") {
       return thunkAPI.rejectWithValue({
-        message: "Please enter your password",
+        message: "Please enter your password"
+      });
+    }
+
+    if (password !== passwordConfirmation) {
```

Comment:
@karkir0003 Here is the logic to check password match.

Reply:
awesome. thanks!